src/server/server_family.cc

```diff
@@ -1820,33 +1825,20 @@ void ServerFamily::ReplicaOfInternal(string_view host, string_view port_sv, Conn
       break;
   };
 
-  VLOG(2) << "Acquire replica lock";
+  // If the replication attempt failed, clean up global state. The replica should have stopped
+  // internally.
   lk.lock();
-
-  // Since we released the replication lock during Start(..), we need to check if this still the
-  // last replicaof command we got. If it's not, then we were cancelled and just exit.
-  if (replica_ == new_replica) {
-    if (ec) {
-      service_.SwitchState(GlobalState::LOADING, GlobalState::ACTIVE);
-      replica_->Stop();
-      replica_.reset();
-    }
-    bool is_master = !replica_;
-    pool.AwaitFiberOnAll(
-        [&](util::ProactorBase* pb) { ServerState::tlocal()->is_master = is_master; });
-  } else {
-    new_replica->Stop();
+  if (ec && replica_ == new_replica) {
+    service_.SwitchState(GlobalState::LOADING, GlobalState::ACTIVE);
+    SetMasterFlagOnAllThreads(false);
+    replica_.reset();
   }
 }
 
 void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
   string_view host = ArgS(args, 0);
   string_view port = ArgS(args, 1);
 
-  // don't flush if input is NO ONE
-  if (!IsReplicatingNoOne(host, port))
-    Drakarys(cntx->transaction, DbSlice::kDbAll);
```

Comment:
you removed the call to Drakarys here but I dont see it inside ReplicaOfInternal

Reply:
that's what I asked above, we already have a flushall when we start full sync, do we need another one here?

Reply:
We do flushall on right before fullsync starts, what if we start replicaof and some of the stages before fullsync starts fails, should we flush the db in this case? If the answer is no than we can remove this flushall , otherwise we should keep it

Reply:
Ok, I'll keep it